src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/ISimdVector_2.cs

```diff
@@ -0,0 +1,450 @@
+
+            if ((uint)(startIndex) >= (uint)(destination.Length))
+            {
+                ThrowHelper.ThrowStartIndexArgumentOutOfRange_ArgumentOutOfRange_IndexMustBeLess();
+            }
+
+            if ((destination.Length - startIndex) < TSelf.Count)
+            {
+                ThrowHelper.ThrowArgumentException_DestinationTooShort();
+            }
+
+            ref byte address = ref Unsafe.As<T, byte>(ref MemoryMarshal.GetArrayDataReference(destination));
+            Unsafe.WriteUnaligned(ref Unsafe.Add(ref address, startIndex), vector);
+        }
+
+        /// <summary>Copies a vector to a given span.</summary>
+        /// <param name="vector">The vector to be copied.</param>
+        /// <param name="destination">The span to which the <paramref name="vector" /> is copied.</param>
+        /// <exception cref="ArgumentException">The length of <paramref name="destination" /> is less than <see cref="Count" />.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        static virtual void CopyTo(TSelf vector, Span<T> destination)
+        {
+            if (!TSelf.IsSupported)
+            {
+                ThrowHelper.ThrowNotSupportedException(ExceptionResource.Arg_TypeNotSupported);
+            }
+
+            if ((uint)(destination.Length) < (uint)(TSelf.Count))
+            {
+                ThrowHelper.ThrowArgumentException_DestinationTooShort();
+            }
+
+            ref byte address = ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(destination));
+            Unsafe.WriteUnaligned(ref address, vector);
+        }
+
+        /// <summary>Creates a new vector with all elements initialized to the specified value.</summary>
+        /// <param name="value">The value that all elements will be initialized to.</param>
+        /// <returns>A new vector with all elements initialized to <paramref name="value" />.</returns>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        static abstract TSelf Create(T value);
+
+        /// <summary>Creates a new vector from a given array.</summary>
+        /// <param name="values">The array from which the vector is created.</param>
+        /// <returns>A new vector with its elements set to the first <see cref="Count" /> elements from <paramref name="values" />.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">The length of <paramref name="values" /> is less than <see cref="Count" />.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        /// <exception cref="NullReferenceException"><paramref name="values" /> is <c>null</c>.</exception>
+        static virtual TSelf Create(T[] values)
```

Comment:
These interfaces will be "more similar to" `Vector<T>` due to it being an abstraction and not "fixed sized".

This means you can't do something like `Create(float e0, float e1, float e2, float e3)` and that may be "restrictive" to some scenarios.

The "workaround" would be for users to pass in such values to the generic implementation (e.g. `SimdImpl<Vector128<TValue>(..., Vector128.Create(...))`)